### DISCUSSION TITLE
Is it possible to use smart pointers ?

### DISCUSSION THREAD

**Discussion by tangosd** [mono/CppSharp#1790] [Q&A] [answered]
I am trying to generate wrapped code for a function returning a shared_ptr. however CppSharp does not produce it, and I can see in the logs the line :
`
Function 'CreateBaseTest' was ignored due to ignored return decl
`
Here is the header file:

```
class BaseTest
{
public:
    std::string Foo();
};


__declspec(dllexport)  std::shared_ptr<BaseTest>  CreateBaseTest();
```

**Comment by tritao** [comment, marked as answer]
This is not supported at the moment, would be a great feature to have but I haven't really needed it and so far no one has stepped up to the plate to implement it :disappointed:
